version: "3.9"
services:
  influxdb:
    image: "influxdb:1.8"
    restart: always
    env_file:
      - ./influxdb-provisioning/influxdb.env
    ports:
      - "8086:8086"
    volumes:
      - type: volume
        source: influxdb
        target: /var/lib/influxdb
      - ./influxdb-provisioning/:/etc/influxdb

  influxdb-config:
    image: "influxdb:1.8"
    env_file:
      - ./influxdb-provisioning/influxdb.env
    command: influx -host influxdb -execute 'CREATE DATABASE pi'
    depends_on:
      - influxdb
    restart: on-failure
  grafana:
    image: grafana/grafana:9.3.0-beta1
    restart: always
    ports:
      - "3000:3000"
    volumes:
      - grafana:/var/lib/grafana
      - ./grafana-provisioning/:/etc/grafana/provisioning
    depends_on:
      - influxdb
    environment:
      - GF_SECURITY_ADMIN_USER=pi
      - GF_SECURITY_ADMIN_PASSWORD=SecurePassword2137
      - GF_PATHS_CONFIG=/etc/grafana/provisioning/config.ini

volumes:
  influxdb:
  grafana:
